### ISSUE TITLE
How to use llama2.c for classification or regression

### ISSUE THREAD

**Issue by thuanvh** [karpathy/llama2.c#203]
Hello, I am newbie for llama2.c. Could you tell me how to use llama2 for regression?
Thank you

**Comment by RahulSChand** [comment]
Do you want to train it on a dataset you have or use already existing weights for some task & just inference?

If you want to train yourself

1. Change `train.py` for your own dataset, changes here depend on your dataset.

2. Change `self.output` in this [line](https://github.com/karpathy/llama2.c/blob/ce05cc28cf1e3560b873bb21837638a434520a67/model.py#L198) to `self.output = nn.Linear(params.dim, 1, bias=False)` for regression & to `self.output = nn.Linear(params.dim, num_classes, bias=False)`  for classification

If you just want to change inference
3. Change `run.c` to remove while loop & just do forward pass once for regression or classification. This [line](https://github.com/karpathy/llama2.c/blob/ce05cc28cf1e3560b873bb21837638a434520a67/run.c#L547).

**Comment by JonathanSum** [comment]
Hi. I am not sure. Llama2 is a language model. I don't understand why you use it for regression problem.

**Comment by karpathy** [comment]
You'd have to gut the repo, which requires PyTorch knowhow. But technically as long as the input is text, the output can be regression by changing the last layer to have a single output and then put a regression loss on it, like L2 or Huber loss or something. It's a non-trivial but possible change. We don't care much about that in this repo so closing.
